### DISCUSSION TITLE
Pick which state should be stored in AsyncStorage?

### DISCUSSION THREAD

**Discussion by midrizi** [infinitered/ignite#2266] [Q&A] [answered]
Is there a way to pick which state should/shouldn't be stored in AsyncStorage?

For example there are certain things that I don't want to be stored since they break the flow of the App

**Comment by midrizi** [comment, marked as answer]
I solved this by using `lodash/pick` function.

Doing the following in `setupRootStore`:
```ts
import pick from 'lodash/pick'

_dipsoser = onSnapshot(rootStore, (snapshot) => {
  return storage.save(ROOT_STATE_STORAGE_KEY, pick(snapshot, ['episodeStore.favoritesOnly']))
})
```

**Reply by wiktorkaplan** [reply]
Hi, this solution works but is not perfect. Every time any state changes, storage.save is called even if the value stored there (in your case episodeStore.favoritesOnly) does not change. Does anyone have any idea how to improve this?
